bin/tartufo-shim.js

```diff
@@ -24,14 +24,22 @@ async function main() {
   if (!tartufo) {
     console.error(chalk.white.bgRed("Tartufo not found locally or globally!"));
     console.error(chalk`We recommend running {cyan npx tartufo-helper doctor} in this project to diagnose this issue.`);
-    process.exit(1);
+    process.exit(0);
   }
 
   const [, , ...args] = process.argv;
 
   debug(chalk`{dim.white ${tartufo} ${args.join(" ")}}`);
   const subprocess = spawn(tartufo, args, { stdio: [process.stdin, process.stdout, process.stderr] });
-  subprocess.on("close", code => process.exit(code));
+  subprocess.on("error", err => {
+    console.error(chalk.white.bgRed(`tartufo failed to run: ${err}`));
+  });
+  subprocess.on("close", code => {
+    if (Number(code) !== 0) {
+      console.error(chalk.white.bgRed(`tartufo failed to run and exited with a ${code}`));
+    }
```

Comment:
We probably actually do want to expose the error code if Tartufo legitimately ran AND it exited with a failure. In this scenario Tartufo detected a secret and we do want to prevent a commit.